From patchwork Sun Jul  5 22:59:51 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Cathy Zhang <cathy.zhang@intel.com>
X-Patchwork-Id: 11644753
Return-Path: <SRS0=HSsq=AQ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 9819D1709
	for <patchwork-kvm@patchwork.kernel.org>;
 Sun,  5 Jul 2020 23:05:22 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 811EC20672
	for <patchwork-kvm@patchwork.kernel.org>;
 Sun,  5 Jul 2020 23:05:22 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1728200AbgGEXFP (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Sun, 5 Jul 2020 19:05:15 -0400
Received: from mga03.intel.com ([134.134.136.65]:22784 "EHLO mga03.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1727894AbgGEXFO (ORCPT <rfc822;kvm@vger.kernel.org>);
        Sun, 5 Jul 2020 19:05:14 -0400
IronPort-SDR: 
 VOVMWviFO37RDzzpSM0bmV6LXAaK9v2Z81m68Jc3Nx6KeF9KLxvR9cdtLo5DWGVnh41Ajs/TlE
 oLhhhX/IjmLA==
X-IronPort-AV: E=McAfee;i="6000,8403,9673"; a="147344300"
X-IronPort-AV: E=Sophos;i="5.75,317,1589266800";
   d="scan'208";a="147344300"
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga001.fm.intel.com ([10.253.24.23])
  by orsmga103.jf.intel.com with ESMTP/TLS/ECDHE-RSA-AES256-GCM-SHA384;
 05 Jul 2020 16:05:14 -0700
IronPort-SDR: 
 9LV4qFlyhOvhAOBOvB5Uq/5HXG6Xbdx9cDw/YOwQc5uGholCwoOrqfv5Uv06ywx+SbcBUciWWM
 8BbybVNyMZIA==
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.75,317,1589266800";
   d="scan'208";a="388043373"
Received: from unknown (HELO localhost.localdomain.bj.intel.com)
 ([10.238.156.127])
  by fmsmga001.fm.intel.com with ESMTP; 05 Jul 2020 16:05:10 -0700
From: Cathy Zhang <cathy.zhang@intel.com>
To: kvm@vger.kernel.org, linux-kernel@vger.kernel.org, x86@kernel.org
Cc: pbonzini@redhat.com, sean.j.christopherson@intel.com,
        vkuznets@redhat.com, wanpengli@tencent.com, jmattson@google.com,
        joro@8bytes.org, tglx@linutronix.de, mingo@redhat.com,
        bp@alien8.de, hpa@zytor.com, ricardo.neri-calderon@linux.intel.com,
        kyung.min.park@intel.com, Cathy Zhang <cathy.zhang@intel.com>
Subject: [PATCH 1/2] x86: Expose SERIALIZE for supported cpuid
Date: Mon,  6 Jul 2020 06:59:51 +0800
Message-Id: <1593989992-10019-2-git-send-email-cathy.zhang@intel.com>
X-Mailer: git-send-email 1.8.3.1
In-Reply-To: <1593989992-10019-1-git-send-email-cathy.zhang@intel.com>
References: <1593989992-10019-1-git-send-email-cathy.zhang@intel.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

SERIALIZE instruction is supported by intel processors,
like Sapphire Rapids. Expose it in KVM supported cpuid.

The SERIALIZE enumeration kernel patch link is as follows:
https://lore.kernel.org/patchwork/patch/1254757/

Signed-off-by: Cathy Zhang <cathy.zhang@intel.com>
---
 arch/x86/kvm/cpuid.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/x86/kvm/cpuid.c b/arch/x86/kvm/cpuid.c
index 8a294f9..e603aeb 100644
--- a/arch/x86/kvm/cpuid.c
+++ b/arch/x86/kvm/cpuid.c
@@ -341,7 +341,8 @@ void kvm_set_cpu_caps(void)
 	kvm_cpu_cap_mask(CPUID_7_EDX,
 		F(AVX512_4VNNIW) | F(AVX512_4FMAPS) | F(SPEC_CTRL) |
 		F(SPEC_CTRL_SSBD) | F(ARCH_CAPABILITIES) | F(INTEL_STIBP) |
-		F(MD_CLEAR) | F(AVX512_VP2INTERSECT) | F(FSRM)
+		F(MD_CLEAR) | F(AVX512_VP2INTERSECT) | F(FSRM) |
+		F(SERIALIZE)
 	);
 
 	/* TSC_ADJUST and ARCH_CAPABILITIES are emulated in software. */

From patchwork Sun Jul  5 22:59:52 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Cathy Zhang <cathy.zhang@intel.com>
X-Patchwork-Id: 11644755
Return-Path: <SRS0=HSsq=AQ=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id B53B614B7
	for <patchwork-kvm@patchwork.kernel.org>;
 Sun,  5 Jul 2020 23:05:22 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 9D64E20672
	for <patchwork-kvm@patchwork.kernel.org>;
 Sun,  5 Jul 2020 23:05:22 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1728250AbgGEXFS (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Sun, 5 Jul 2020 19:05:18 -0400
Received: from mga03.intel.com ([134.134.136.65]:22784 "EHLO mga03.intel.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1728231AbgGEXFS (ORCPT <rfc822;kvm@vger.kernel.org>);
        Sun, 5 Jul 2020 19:05:18 -0400
IronPort-SDR: 
 fy2sIY0UDSxGiICBF+b/VJ7WyuVJie2pGW7zH87KKbmwA8dMac7hiwZBeIKjvgTO3AzEizedzx
 aSZsC9qIASaQ==
X-IronPort-AV: E=McAfee;i="6000,8403,9673"; a="147344307"
X-IronPort-AV: E=Sophos;i="5.75,317,1589266800";
   d="scan'208";a="147344307"
X-Amp-Result: SKIPPED(no attachment in message)
X-Amp-File-Uploaded: False
Received: from fmsmga001.fm.intel.com ([10.253.24.23])
  by orsmga103.jf.intel.com with ESMTP/TLS/ECDHE-RSA-AES256-GCM-SHA384;
 05 Jul 2020 16:05:17 -0700
IronPort-SDR: 
 Sjl7u5d9r8as9U9VZQPFh0vjdZ1ZJsVWPShKTMBDznonf7Qf6SEJER6HYZULSoPnmrbllMRD6y
 1zafDSHN2xfw==
X-ExtLoop1: 1
X-IronPort-AV: E=Sophos;i="5.75,317,1589266800";
   d="scan'208";a="388043382"
Received: from unknown (HELO localhost.localdomain.bj.intel.com)
 ([10.238.156.127])
  by fmsmga001.fm.intel.com with ESMTP; 05 Jul 2020 16:05:14 -0700
From: Cathy Zhang <cathy.zhang@intel.com>
To: kvm@vger.kernel.org, linux-kernel@vger.kernel.org, x86@kernel.org
Cc: pbonzini@redhat.com, sean.j.christopherson@intel.com,
        vkuznets@redhat.com, wanpengli@tencent.com, jmattson@google.com,
        joro@8bytes.org, tglx@linutronix.de, mingo@redhat.com,
        bp@alien8.de, hpa@zytor.com, ricardo.neri-calderon@linux.intel.com,
        kyung.min.park@intel.com, Cathy Zhang <cathy.zhang@intel.com>
Subject: [PATCH 2/2] x86: Expose TSX Suspend Load Address Tracking
Date: Mon,  6 Jul 2020 06:59:52 +0800
Message-Id: <1593989992-10019-3-git-send-email-cathy.zhang@intel.com>
X-Mailer: git-send-email 1.8.3.1
In-Reply-To: <1593989992-10019-1-git-send-email-cathy.zhang@intel.com>
References: <1593989992-10019-1-git-send-email-cathy.zhang@intel.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

TSX Suspend Load Address Tracking is supported by intel processors,
like Sapphire Rapids. Expose it in KVM supported cpuid.

The associated kernel enumeration patches link is as follows:
https://lore.kernel.org/patchwork/patch/1254756/

Signed-off-by: Cathy Zhang <cathy.zhang@intel.com>
---
 arch/x86/kvm/cpuid.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kvm/cpuid.c b/arch/x86/kvm/cpuid.c
index e603aeb..dcf48cc 100644
--- a/arch/x86/kvm/cpuid.c
+++ b/arch/x86/kvm/cpuid.c
@@ -342,7 +342,7 @@ void kvm_set_cpu_caps(void)
 		F(AVX512_4VNNIW) | F(AVX512_4FMAPS) | F(SPEC_CTRL) |
 		F(SPEC_CTRL_SSBD) | F(ARCH_CAPABILITIES) | F(INTEL_STIBP) |
 		F(MD_CLEAR) | F(AVX512_VP2INTERSECT) | F(FSRM) |
-		F(SERIALIZE)
+		F(SERIALIZE) | F(TSX_LDTRK)
 	);
 
 	/* TSC_ADJUST and ARCH_CAPABILITIES are emulated in software. */
